src/identity-handler.js

```diff
@@ -21,7 +21,26 @@ For more userIdentity types, see http://docs.mparticle.com/developers/sdk/javasc
 function IdentityHandler(common) {
     this.common = common || {};
 }
-IdentityHandler.prototype.onUserIdentified = function(mParticleUser) {};
+IdentityHandler.prototype.onUserIdentified = function(mParticleUser) {
+    var mPUser = mParticleUser.getUserIdentities().userIdentities;
+    var userId = '';
+
+    switch (this.common.settings.userIdentificationType) {
```

Comment:
I'm not sure where you're setting `userIdentificationType` or `settings` for that matter. You should probably have a safety check before this line to make sure `settings` and `userIdentificationType` are not null.

Reply:
Is it possible for either of those to be null while `mParticleUser.getUserIdentities().userIdentities` returns valid identities? if so is it better to get the identity type from `mParticleUser.getUserIdentities().userIdentities` somehow?

Reply:
I realize there is a bug in the file and put up a PR with a few [bugfixes](https://github.com/mparticle-integrations/mparticle-javascript-integration-example/pull/21) in our integration-example repo, but specifically it's this [commit](https://github.com/mparticle-integrations/mparticle-javascript-integration-example/pull/21/commits/b2c5294b430941b14047cafa4908e48f022d292c).  

once this is merged, you can rebase the updates.

you can then follow branch's lead on this line [here](https://github.com/BranchMetrics/mparticle-javascript-integration-branch/blob/master/src/initialization.js#L4).  you set the forwarder settings onto the `common` module, which is then available here via `this.common.settings` as they've done.

i'll ping this thread once the PR above is merged

Reply:
another thing - you said you took branch's lead on this. they specifically called out a `userIdentificaitonType` setting that is available in Branch.  Does userleap have something like this?  or is there just a userId.  i assume that would match to `customerId` but since i'm not familiar with your platform, some guidance would be helpful.

it looks like the settings discussed were only `apiKey` and `environmentId`.  please confirm.

Reply:
btw the above PR was merged so you could rebase it in

Reply:
Simplifying the implementation, let me know if this is valid